src/network/raknet/server/RakNetServer.ts

```diff
@@ -0,0 +1,134 @@
+import { RAKNET_PROTOCOL } from "./RakNetCostants";
+import OpenConnectionRequestTwo from "../protocol/OpenConnectionRequestTwo";
+import OpenConnectionReplyTwo from "../protocol/OpenConnectionReplyTwo";
+import NetworkQueueHandler from "../handler/NetworkQueueHandler";
+
+interface RakNetServerData extends InetAddressData {
+    /**
+     * Socket interface type (IPv4/IPv6).
+     */
+    type: "udp4" | "udp6";
+    /**
+     * Maximum transfer unit (buffer per packet).
+     */
+    mtu?: number;
+    /**
+     * Maximum client connections.
+     */
+    maxConnections?: number;
+}
+
+export default class RakNetServer implements IRakNetServer {
+    private readonly bindAddress: InetAddress;
+    private readonly socket: DSocket = createSocket("udp4");  
+    private readonly GUID: bigint = randomBytes(8).readBigInt64BE();
+    private clients: Map<String, RakNetConnection> = new Map(); 
+
+    constructor(raknetData: RakNetServerData) {
+        if (raknetData.type == "udp6") {
+            throw new Error("IPv6 is not supported yet");
+        }
+        this.bindAddress = new InetAddress(raknetData);
+    }
+
+    public listen(): void {
+        try {
+            this.socket.bind({
+                address: this.bindAddress.getAddress(),
+                port: this.bindAddress.getPort()
+            }, () => console.log(
+                "RakNet socket listening on port %d", this.bindAddress.getPort()
+            )); // TODO: replace with an actual logger
+        } catch (err) {
+            throw err;
+        }
+
+        this.socket.on("message", async (msg, rinfo) => await this.handleMessage(msg, rinfo));
+    }
+
+    private async handleMessage(msg: Buffer, rinfo: RemoteInfo): Promise<void> {
+        const header = msg.readUInt8();  
+
+        let token = InetAddress.fromRemoteInfo(rinfo).toToken();
+        if (this.clients.has(token)) {
+             
+        } else {
+            this.socket.send(
+                await this.handleOfflineMessage(header, msg, rinfo),
+                rinfo.port, rinfo.address
+            )
+        }
+    }
+
+    public async handleOfflineMessage(header: number, msg: Buffer, rinfo: RemoteInfo): Promise<Buffer> {
+        return await new Promise(resolve => {
+            let decoded, encoded;
+            switch (header) {
+                case RakNetIdentifiers.UNCONNECTED_PING:
+                    decoded = new UnconnectedPing(msg);
+                    decoded.decodeInternal();
+
+                    if (!decoded.isMagicValid()) {
+                        // TODO: skip fake Offline packet...
+                    }
+
+                    encoded = new UnconnectedPong();
+                    encoded.timestamp = decoded.timestamp;
+                    encoded.pongId = this.GUID;
+                    encoded.serverName = 'MCPE;JSRakNet;407;1.16.0;0;5;server_id;JSRakNet;Creative;1;19132;19132;';  // TODO: change with actual MOTD
```

Comment:
In addition to the comment, I believe this should be modifiable (to an extent) by plugins. EG: Ability to change version, protocol, and gamemode, plugins, and player count (for proxy related things that could be created).